### PR TITLE
Moves the React hooks entrypoint from @apollo/client to @apollo/client/react

### PR DESCRIPTION
**Pull request by MrDoomBringer**
See this milestone for more information: https://github.com/apollographql/apollo-client/issues/8190

This PR aims to remove the react entrypoint from `@apollo/client` and have it accessible only through `@apollo/client/react`. 
- Adjusts `@apollo/client` and `@apollo/client/react` accordingly
- Adds an `@apollo/testing/react` entrypoint, used primarily for MockedProvider
- Includes a basic codemod to help move react hook imports out of /client and into client/react
- Includes a basic migration guide, currently only including information about this change
- Updates all AC React docs
### Checklist:

- [X] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [X] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by hwillson** [comment]
Super excited about this! Let us know when you think this is ready for review @MrDoomBringer - thanks!

File: docs/source/migrating/apollo-client-4-migration.mdx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: Migrating to Apollo Client 3.0
```
Comment (alessbell) [inline docs/source/migrating/apollo-client-4-migration.mdx:2]: 😄
```suggestion
title: Migrating to Apollo Client 4.0
```

Comment (MrDoomBringer) [reply docs/source/migrating/apollo-client-4-migration.mdx:2]: Haha there's always something that gets through

**Comment by jpvajda** [comment]
@MrDoomBringer 👋  can this be merged in now?

**Comment by MrDoomBringer** [comment]
@jpvajda Yup! Just waiting for an approval review now :-)

**Pull request merged**